transport/controller.cc

```diff
@@ -181,7 +182,20 @@ future<> controller::start_listening_on_maintenance_socket(sharded<cql_server>&
 
     logger.info("Setting up maintenance socket on {}", socket);
 
-    return listen_on_all_shards(cserver, addr, nullptr, false, _config.rpc_keepalive(), unix_domain_socket_permissions);
+    co_await listen_on_all_shards(cserver, addr, nullptr, false, _config.rpc_keepalive(), unix_domain_socket_permissions);
+
+    if (_config.maintenance_socket_group.is_set()) {
+        struct group *grp;
+        grp = ::getgrnam(_config.maintenance_socket_group().c_str());
+        if (grp) {
+            auto chown_result = ::chown(socket.c_str(), ::geteuid(), grp->gr_gid);
```

Comment:
> Looks like there is a func for that: explain_message_errno_chown

It would need library `libexplain` to be installed. I don't know if it will always be true.

Reply:
then perhaps some documentation and mention maintenance socket in the error message? If user sets this option he needs to run scylla as a part of this group, maybe it's obvious when you read the code, but as an user you get random chown it may be less obvious.

I think the message currently would look similar to:
"Failed to chown /path/cql.m: EPERM: Not owner"

Don't know exactly how strerror output look like in this case. But as you can see there is no link to maintenance socket feature and user may be a bit lost.